archive/app/controllers/ArchiveController.scala

```diff
@@ -31,16 +39,51 @@ class ArchiveController(redirects: RedirectService, renderer: Renderer, val cont
 
   private val redirectHttpStatus = HttpStatus.SC_MOVED_PERMANENTLY
 
-  def lookup(path: String): Action[AnyContent] = Action.async{ implicit request =>
+  // todo: refactor out since this is going to be used in many places
+  def get404Page: String = {
+    val response = ws.url(Configuration.moon.moonEndpoint)
+      .withRequestTimeout(6000.millis)
+      .get()
+      .map((response) => {
+        response.body
+      })
+
+    Await.result(response, 5000.millis)
```

Comment:
Moving that at the calling site would give the caller the ability to decide how long to wait and what to do in case of failure.

Reply:
... and record the number of failures